altair/utils/schemapi.py

```diff
@@ -510,10 +519,16 @@ def _todict(obj: Any, context: dict[str, Any] | None, np_opt: Any, pd_opt: Any)
             for k, v in obj.items()
             if v is not Undefined
         }
-    elif hasattr(obj, "to_dict"):
+    elif (
+        hasattr(obj, "to_dict")
+        and (module_name := obj.__module__)
+        and module_name.startswith("altair")
```

Comment:
What are these additional checks for and why weren't they needed previously?

Reply:
That was added in [`3275c5e` (#3501)](https://github.com/vega/altair/pull/3501/commits/3275c5efa837e52f8d7f40f297cd298879d5b6a7)

> fix: Prevent `to_dict` method being called on `pd.Series`
> AFAIK, this is the intended case for `Parameter | Expression` not for converting arbitrary objects

To add to that, it is because that branch is *before*:

```py
elif _is_iterable(obj, exclude=(str, bytes)):
	return _todict(_from_array_like(obj), context, np_opt, pd_opt)
```

So that code would never be reached for any object like:

```py
from typing import Any, Iterable, Iterator

class SomeObj(Iterable):
    def to_dict(self) -> Iterator[Any]: ...
```

`pd.Series` met these criteria during testing